includes/class-events.php

```diff
@@ -31,13 +31,19 @@ protected function class_init() {
 	 * This also runs before Core has parsed the request and set the \REST_REQUEST constant
 	 */
 	public function prepare_environment() {
-		if ( ! is_rest_endpoint_request( 'run' ) ) {
+		// Limit to plugin's endpoints
+		if ( ! is_rest_endpoint_request( 'list' ) && ! is_rest_endpoint_request( 'run' ) ) {
```

Comment:
Yeah, probably a good idea. This is the first situation where we've needed to check multiple endpoints, but it probably won't be the last given the inconsistent behaviour this PR resolves.